daemon/cluster/executor/container/executor.go

```diff
@@ -154,9 +159,35 @@ func (e *executor) Controller(t *api.Task) (exec.Controller, error) {
 		return newNetworkAttacherController(e.backend, t, e.secrets)
 	}
 
-	ctlr, err := newController(e.backend, t, secrets.Restrict(e.secrets, t))
-	if err != nil {
-		return nil, err
+	var ctlr exec.Controller
```

Comment:
@ehazlett out of curiosity, was it necessary to handle the generic taskspec case in the container executor instead of having a generic executor? Or is the rationale that if we decide to migrate the container spec to generic spec, then the container executor becomes a defacto generic executor? Disregard if this makes no sense.